src/main/java/nl/opengeogroep/safetymaps/server/admin/stripes/FotoActionBean.java

```diff
@@ -0,0 +1,154 @@
+        String location = context.getRequest().getParameter("location");
+        if (filename != null) {
+            try {
+                deleteFromFileSystem(location);
+                DB.qr().update("delete from wfs."+TABLE+" where filename ='" + filename + "'");
+                getContext().getMessages().add(new SimpleMessage("Foto verwijderd."));
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        } else {
+            getContext().getMessages().add(new SimpleMessage("Geen bestandsnaam om te verwijderen!"));
+        }
+        list();
+        return new ForwardResolution(JSP);
+    }
+
+    @DontValidate
+    public Resolution downloadFoto() throws Exception {
+        String filename = context.getRequest().getParameter("filename");
+        String location = context.getRequest().getParameter("location");
+        System.out.println("downloading: " + location);
+        File file = new File(location);
+        Path source = Paths.get(location);
+        context.getResponse().setBufferSize(DEFAULT_BUFFER_SIZE);
+        context.getResponse().setContentType(Files.probeContentType(source));
+        context.getResponse().setHeader("Content-Length", String.valueOf(file.length()));
+        context.getResponse().addHeader("content-disposition", "attachment; filename=" + filename);
+        
+        FileInputStream fis = null;
+        BufferedInputStream input = null;
+        BufferedOutputStream output = null;
+
+        try {
+            fis = new FileInputStream(file);
+            input = new BufferedInputStream(fis, DEFAULT_BUFFER_SIZE);
+            output = new BufferedOutputStream(context.getResponse().getOutputStream(), DEFAULT_BUFFER_SIZE);
+
+            
+            byte[] buffer = new byte[DEFAULT_BUFFER_SIZE];
+            int length;
+            while ((length = input.read(buffer)) > 0) {
+                output.write(buffer, 0, length);
+            }
+        } finally {
+            
+            output.close();
+            input.close();
+        }
+        return null;
+    }
+    
+    @DontValidate
+    public byte[] getImage(String fileName) {
+        byte[] result = null;
+        String fileLocation = fileName;
+        File f = new File(fileLocation);
+        result = new byte[(int) f.length()];
+        try {
+            FileInputStream in = new FileInputStream(fileLocation);
+            in.read(result);
+        } catch (Exception ex) {
+            System.out.println("GET IMAGE PROBLEM :: " + ex);
+            ex.printStackTrace();
+        }
+        return result;
+    }
+    
+    public void deleteFromFileSystem(String path) {
+        try {
+
+            File file = new File(path);
+
+            if (file.delete()) {
+                System.out.println(file.getName() + " is deleted!");
+            } else {
+                System.out.println("Delete operation failed.");
+            }
+        } catch (Exception e) {
```

Comment:
Swallows exception, so when file is not deleted the delete() resolution continues deleting the record